src/payload/payload.ts

```diff
@@ -0,0 +1,306 @@
+import { unwrap } from '../client/intern';
+import { Kwil } from '../client/kwil';
+import { SignerSupplier } from '../core/builders';
+import {
+  BytesEncodingStatus,
+  EnvironmentType,
+  PayloadType,
+  SerializationType,
+} from '../core/enums';
+import { BaseMessage, Message, Msg } from '../core/message';
+import { AllPayloads, UnencodedActionPayload } from '../core/payload';
+import { AnySignatureType, executeSign, Signature, SignatureType } from '../core/signature';
+import { BaseTransaction, Transaction, Txn } from '../core/tx';
+import { base64ToBytes, bytesToBase64 } from '../utils/base64';
+import { sha256BytesToBytes } from '../utils/crypto';
+import { objects } from '../utils/objects';
+import { kwilEncode } from '../utils/rlp';
+import { bytesToHex, stringToBytes } from '../utils/serial';
+import { strings } from '../utils/strings';
+
+export interface PayloadOptions {
+  payload: (() => AllPayloads) | AllPayloads;
```

Comment:
I actually don't think we accept a CB in any of our public APIs anymore, so you can prob change this to just:

```
payload: AllPayloads
```